apps/src/JSInterpreter.js

```diff
@@ -324,6 +326,17 @@ JSInterpreter.prototype.removeStoppedAtBreakpointRowForScope = function (scope)
 };
 
 /**
+ * Determines if the program is done executing.
+ *
+ * @return {boolean} true if program is complete (or an error has occurred).
+ */
+JSInterpreter.prototype.isProgramDone = function () {
+  return this.executionError ||
+      !this.interpreter ||
+      !this.interpreter.stateStack.length;
+};
```

Comment:
I think this is weird today because an execution error doesn't immediately clear out the `stateStack` (which it would in a real JavaScript program).  The caller needs to listen for `onExecutionError` to know not to keep calling `executeInterpreter()`.  Maybe this is as simple as dumping the `stateStack` when an uncaught error occurs?

Until `try`/`catch` is implemented, all errors will be uncaught (and should halt the interpreter).